ngraph_bridge/ie_executable.cc

```diff
@@ -40,12 +40,18 @@ IE_Executable::IE_Executable(shared_ptr<Function> func, string device)
     }
   }
 
+  // A trivial function is one of
+  //  1. constant function (Const -> Result)
+  //  2. identity function (Parameter -> Result)
+  //  3. zero function (* -> Zero)
   NGRAPH_VLOG(2) << "Checking for trivial functions in IE backend";
   bool trivial_fn = true;
   for (auto result : func->get_results()) {
     auto parent = result->input_value(0).get_node_shared_ptr();
+    auto& shape = result->get_shape();
     trivial_fn &= ngraph::is_type<opset::Parameter>(parent) ||
-                  ngraph::is_type<opset::Constant>(parent);
+                  ngraph::is_type<opset::Constant>(parent) ||
+                  count(shape.begin(), shape.end(), 0);
```

Comment:
Why do we have the zero dim check at 2 places?

Reply:
This instance marks the function as a trivial function. When the trivial function is being executed, we handle all three cases separately and have to check what type of trivial function it is.